ros2param/ros2param/verb/dump.py

```diff
@@ -93,17 +80,24 @@ def main(self, *, args):  # noqa: D102
                     'Exception while calling list_parameters service of node '
                     f"'{node_name.full_name}': {e}", file=sys.stderr)
                 return
+            parameter_names = sorted(response.result().result.names)
 
-            response = response.result().result.names
-            response = sorted(response)
-            parameter_values = self.get_parameter_values(node, absolute_node_name, response)
-            if parameter_values is None:
+            # retrieve parameter values
+            try:
+                response = call_get_parameters(
+                    node=node, node_name=absolute_node_name, parameter_names=parameter_names)
+            except RuntimeError as e:
                 print(
                     'Exception while calling get_parameters service of node '
                     f"'{node_name.full_name}': {e}", file=sys.stderr)
+            if response.values is None:
+                # no parameters available
+                print(f"'{node_name.full_name}' does not possess any parameters.")
```

Comment:
Should we print this to user? i do not have strong opinion here though.

- keep this printing so that user can understand there are no parameter available with this node.
- no print message here, so that nothing comes up in the terminal.
- we do not return here, so that it outputs the yaml `[node_name.full_name]['ros__parameters']` only.

i think printing message would be the most comprehensive for user perspective.

Reply:
It's a tough one to answer.  I think it kind of depends on whether we intend this to be used as an end-user interface, or as part of a pipeline.  For an end-user interface, I think printing "<node> has no parameters" is best.  For a pipeline, I think the empty YAML is best.  I guess we could add a command-line option to control it, but that doesn't feel great either.

I guess my personal preference would be to remove the return and print the empty YAML.  But I'd also like to hear opinions from @ros2/team .

Reply:
I agree with @clalancette that an empty YAML is fine. This command should only write a yaml file to stdout.

Reply:
How about printing the error message to STDERR and an empty yaml on STDOUT? Then it can still be used as a pipeline and have a human readable error message.

Reply:
thank you very much for the comments.

i decided to take `empty YAML` process. Having no parameters is NOT the error, that is just one of the success case. for consistent behavior, i think we can pass through this case and proceed to print the (almost) empty YAML.

Note: the question is `can we create a node with no parameters available?` the short answer is yes but only if user application creates the node with `rcl` interfaces. there are 2 default (aside from rclcpp QoS parameters) parameters available when it creates the node, that are `use_sim_time` and `start_type_description_service`. `use_sim_time` can be undeclared because it is not `read_only` parameter. but we cannot undeclare `start_type_description_service` parameter because it is declared with `read_only` parameter by `rclcpp` and `rclpy`. after all, this `no parameters available` situation is so unlikely to happen. because of this, i ended up not having the test case for this either.